pkg/askpass/credentials.go

```diff
@@ -0,0 +1,148 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package askpass
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"cloud.google.com/go/auth"
+	"cloud.google.com/go/auth/credentials"
+	"k8s.io/klog/v2"
+)
+
+// DefaultSourceScopes returns the scopes needed to fetch source from CSR and SSM.
+func DefaultSourceScopes() []string {
+	return []string{"https://www.googleapis.com/auth/cloud-platform"}
+}
+
+// IsCredentialsNotFoundError returns true if an error from
+// credentials.DetectDefault indicates that no credentials are configured.
+func IsCredentialsNotFoundError(err error) bool {
+	if err == nil {
+		return false
+	}
+	// Check the error prefix, because credentials.DetectDefault doesn't use a
+	// custom error type.
+	// https://github.com/googleapis/google-cloud-go/blob/auth/v0.12.0/auth/credentials/detect.go#L108
+	return strings.HasPrefix(err.Error(), "credentials: could not find default credentials.")
+}
+
+// CredentialProvider specifies an interface for anything that can return
+// credentials.
+type CredentialProvider interface {
```

Comment:
The CredentialProvider is used by OCI, Helm and Otel-collector. Should it be relocated to a separate package rather than askpass. The askpass is more specific to Git.

Reply:
moved to `pkg/auth`